DockerReadme.md

```diff
@@ -0,0 +1,70 @@
+## Configuring the docker environment.
+
+## For Linux you can install docker directly, but for mac and windows a linux vm is required to host the docker containers.
+
+Installation instructions for mac and windows can be found;
+
+here
+
+https://docs.docker.com/installation/mac/
+
+and here
+
+https://docs.docker.com/installation/windows/
+
+## Building the docker container. (This assumes boot2docker is running and working as per the instructions above if you are
+on mac or windows.)
+
+ # background
+Docker requires a container to be built in order for it to run our code. Our current scenario downloads a base image
+that is listed in the Dockerfile. That image contains installations of node, gulp and bower. Once that images is
+downloaded it then runs apt-get to install updates to the packages so we are using the latest versions. (This may not be
+the desired behavior at some point, but during development it is how things are working.)
+
+Once the container has been updated to use the newest versions of the software we need, it will run through the build
+process using npm install and gulp, and when that is done the process is complete.
+
+Note: This seems to have issues on slower network connections, and can take a few minutes to happen.
+
+ # build
+run the following command from a
```

Comment:
Add note that if you have previously built outside of Docker, you need to remove `node_modules` due to platform-specific builds.